### ISSUE TITLE
Question about using this plugin with react

### ISSUE THREAD

**Issue by sara3988** [nagix/chartjs-plugin-streaming#54]
I am trying to build a prototype with react using this plugin. I return the Line chart as below
`<Line ref="chart" data={data} />`

**data** as below:
![image](https://user-images.githubusercontent.com/19271236/49072466-9364aa00-f230-11e8-8b5c-328c1e6d7fcf.png)

I do see the chart being rendered, but there is no real-time data, it seems static. 
![image](https://user-images.githubusercontent.com/19271236/49071866-34526580-f22f-11e8-88c7-d8f55e3bc959.png)

I see the varying values in onRefresh function(Using console logs)
![image](https://user-images.githubusercontent.com/19271236/49072026-98752980-f22f-11e8-93bb-f8ba4d46f542.png)

I can see I am missing or over-doing something, but unable to figure out what that would be. Any help is highly appreciated, thanks!

**Comment by gj1118** [comment]
@sara3988 , Can you please let me know if you got it working .

**Comment by gj1118** [comment]
@sara3988 thanks for the pointer. Can you please share how you are using `dataset.data.shift()`, please ?  I find it bizarre that the dataset is getting updated but nothing is getting plotted for me. 

Here is a screenshot from my console window 
https://cl.ly/55ddee32e0f7

Thanks

**Comment by sara3988** [comment]
`chart.config.data.datasets.forEach(function(dataset) {
        if (dataset.data.length > 12) {
          dataset.data.shift();
        }
    		dataset.data.push({
    			x: Date.now(),
    			y: <<value>>
    		});
        chart.update();
    	});`
I don't think this is the right way to do it, but I was just trying out possibilities.

On a side note, your data from 0-7 : do not seem to have x and y values, probably this is the problem?

**Comment by gj1118** [comment]
Thanks. 
I have almost the same code , but for me, nothing is getting plotted. Please see 

![image](https://user-images.githubusercontent.com/473101/49139549-154fe400-f2c0-11e8-87ba-0a8a90463d95.png)

That said, I am wondering if x and y values are not present, they are not getting plotted, but the ones which are present, should get plotted,no ?

Thanks once again for your time and patience.

**Comment by sara3988** [comment]
Can you try **removing duration: 0** from update method and see if it works? (Duration of the chart in milliseconds (how much time of data it will show)).

**onRefresh**: According to dicumentation : In this callback function, you can add data into the existing data array as usual, but you don't need to call the update function as it is called internally. But nothing happens for me if I do not call update() 

I see you have **paused the streaming**. (If set to true, scrolling stops. Note that onRefresh callback is called even when this is set to true). Try removing this?

Also, should "x" always be Date.now() or can it be any number? You "x" values do not seem to follow a pattern, and is kind of random. What if for testing purpose, you change it to `Date.now()`

**Comment by gj1118** [comment]
DId all those things, but it did not work. Also when I try adding x, y values to my dataset, i get the following error 

![image](https://user-images.githubusercontent.com/473101/49141338-5e099c00-f2c4-11e8-86dc-e28f97b957a6.png)

**Comment by sara3988** [comment]
Should wait for the developers to respond.

**Comment by nagix** [comment]
I think `options` needs to be specified as a tag attribute, not a part of `data`. As seen in your chart image, no options are reflected in the chart because of this.
```html
<Line ref="chart" data={data} options={options} />
```
```javascript
data = {
    "datasets": [
    ...
};
options = {
    "title": {
    ...
};
```

See also a React example in the 'React' tab at https://nagix.github.io/chartjs-plugin-streaming/

**Comment by sara3988** [comment]
That's right, I guess seeing multiple documents at the same time kind of put me off the chart. That helped. Thanks a lot. This issue can be closed.
